types/daterangepicker/index.d.ts

```diff
@@ -18,15 +19,15 @@ declare global {
 declare const daterangepicker: daterangepicker.DateRangePicker;
 
 declare namespace daterangepicker {
-    type DataRangePickerCallback = (start?: string | Date | moment.Moment, end?: string | Date | moment.Moment, label?: string) => any;
+    type DataRangePickerCallback = (start: moment.Moment, end: moment.Moment, label: string) => any;
```

Comment:
Changes for `start` and `end` seem right. However, `label` can be `null` (see link), for that reason I would leave it optional tbh.
https://github.com/dangrossman/bootstrap-daterangepicker/blob/9670cb08601d0fd7d9fa39734ab6f9f1ef354e8c/daterangepicker.js#L1399

Reply:
It would be more correct not to make it optional (because it is always sent) and instead to change it to `label: string | null`. I will update this.

Note that a function that only takes one or two of the parameters will still match even if the args are optional. Making something optional matters when the function is invoked, not when it's defined.

Reply:
Good point 👍  I agree, `label: string | null`, is better.